quandl/model/data.py

```diff
@@ -6,18 +5,30 @@
 
 
 class Data(DataListOperation, DataMixin, ModelBase):
+
     def __init__(self, data, **options):
         self.meta = options['meta']
-        if 'column_names' in options['meta'].keys():
-            the_list = [Util.methodize(x) for x in options['meta']['column_names']]
-            converted_column_names = list(the_list)
+        self._raw_data = Util.convert_to_dates(data)
+
+        # Optimization for when a list of data points are created from a dataset (via the model_list class)
+        if 'converted_column_names' in options.keys():
+            self._converted_column_names = options['converted_column_names']
+
+    # Need to override data_fields incase the way the Data class was populated that it did not contain a
+    # converted_column_names option passed in when it was created.
+    def data_fields(self):
+        if not self._converted_column_names and self.meta:
```

Comment:
does this introduce a bug? once `self._converted_column_names` is set, it will no longer go into this if statement, and it will go into the `else`? which returns `[]` empty aray?